user_data/hyperopts/test_hyperopt.py

```diff
@@ -0,0 +1,283 @@
+# pragma pylint: disable=missing-docstring, invalid-name, pointless-string-statement
+
+import talib.abstract as ta
+from pandas import DataFrame
+from typing import Dict, Any, Callable
+from functools import reduce
+from math import exp
+
+import numpy
+import talib.abstract as ta
+from hyperopt import STATUS_FAIL, STATUS_OK, Trials, fmin, hp, space_eval, tpe
+
+import freqtrade.vendor.qtpylib.indicators as qtpylib
+from freqtrade.indicator_helpers import fishers_inverse
+from freqtrade.optimize.interface import IHyperOpt
+
+
+# Update this variable if you change the class name
+class_name = 'TestHyperOpt'
+
+
+# This class is a sample. Feel free to customize it.
+class TestHyperOpt(IHyperOpt):
+    """
+    This is a test hyperopt to inspire you.
+    More information in https://github.com/gcarq/freqtrade/blob/develop/docs/hyperopt.md
+
+    You can:
+    - Rename the class name (Do not forget to update class_name)
+    - Add any methods you want to build your hyperopt
+    - Add any lib you need to build your hyperopt
+
+    You must keep:
+    - the prototype for the methods: populate_indicators, indicator_space, buy_strategy_generator,
+    roi_space, generate_roi_table, stoploss_space
+    """
+
+    @staticmethod
+    def populate_indicators(dataframe: DataFrame) -> DataFrame:
+        """
+        Adds several different TA indicators to the given DataFrame
+
+        Performance Note: For the best performance be frugal on the number of indicators
+        you are using. Let uncomment only the indicator you are using in your strategies
+        or your hyperopt configuration, otherwise you will waste your memory and CPU usage.
+        """
+
+        # Momentum Indicator
+        # ------------------------------------
+
+        # ADX
+        dataframe['adx'] = ta.ADX(dataframe)
+
+        """
+        # Awesome oscillator
```

Comment:
why leave these in the test if they are commented out?

Reply:
Similar to ``user_data/strategies/test_strategy``, to provide examples to the user of what can be included in their custom hyperopt